pina/solvers/garom.py

```diff
@@ -0,0 +1,256 @@
+""" Module for PINN """
+import torch
+try:
+    from torch.optim.lr_scheduler import LRScheduler  # torch >= 2.0
+except ImportError:
+    from torch.optim.lr_scheduler import _LRScheduler as LRScheduler # torch < 2.0
+
+from torch.optim.lr_scheduler import ConstantLR
+from .solver import SolverInterface
+from ..utils import check_consistency
+from ..loss import LossInterface, LpLoss
+from torch.nn.modules.loss import _Loss
+
+
+class GAROM(SolverInterface):
+    """
+    GAROM solver class. This class implements Generative Adversarial
+    Reduced Order Model solver, using user specified ``models`` to solve
+    a specific order reduction``problem``. 
+
+    .. seealso::
+
+        **Original reference**: Coscia, D., Demo, N., & Rozza, G. (2023).
+        Generative Adversarial Reduced Order Modelling. 
+        arXiv preprint arXiv:2305.15881.
+        <https://doi.org/10.48550/arXiv.2305.15881>`_.
+    """
+    def __init__(self,
+                 problem,
+                 generator,
+                 discriminator,
+                 extra_features=None,
+                 loss = LpLoss(p=1),
```

Comment:
https://stackoverflow.com/questions/32939159/create-object-with-attributes-that-have-default-values-python

Reply:
solved in [5998ac7](https://github.com/mathLab/PINA/pull/133/commits/5998ac76f0aed8c8e0b230c081fd3d4d358e8470)